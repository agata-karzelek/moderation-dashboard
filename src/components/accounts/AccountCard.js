import React from "react";
import { MenuItem, Avatar, Grid, Typography } from "@material-ui/core";
import CheckCircleIcon from "@material-ui/icons/CheckCircle";
import { useStyles } from "../../style/accountCard";
import { capitalizeFirstCharacter, avatarStyle, selectedAccountsFromLS } from "../../utils/helpers";

const AccountCard = ({ account, handleClick, index }) => {
  const classes = useStyles();

  return (
    <>
      <MenuItem testid={`${index}`} onClick={() => handleClick(account)}>
        <Grid container>
          <Grid item xs={2} sm={2}>
            <Avatar className={classes.avatar} style={avatarStyle(index)} variant="square">
              <Typography testid="capitalChar" className={classes.avatarText}>
                {account && account.email
                  ? capitalizeFirstCharacter(account.email.slice(0, 2))
                  : null}
              </Typography>
            </Avatar>
          </Grid>
          <Grid item xs={9} sm={9}>
            <Typography testid="email" noWrap className={classes.email}>
              {account.email}
            </Typography>
          </Grid>
          <Grid item xs={1} sm={1}>
            <CheckCircleIcon
              style={avatarStyle(account.email, selectedAccountsFromLS)}
              className={classes.checkIcon}
            />
          </Grid>
        </Grid>
      </MenuItem>
    </>
  );
};

export default AccountCard;
