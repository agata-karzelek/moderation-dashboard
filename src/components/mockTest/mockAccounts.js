export const mockAccounts = [
  {
    created: null,
    dates: {
      company: {
        created: 1619760528,
        modified: null,
      },
      product_of_interest: {
        created: 1619760609,
        modified: null,
      },
      use_case: {
        created: 1619760609,
        modified: null,
      },
    },
    email: "abc@gmail.com",
    id: 572238,
    internal_cipher_key: ")[t*)u^>[\\\\t{7pi>./t2@,@-9}\\ys31",
    mau_pricing_plan: {
      pricing_plan: {
        plan_type: "mau_t1",
      },
    },
    modified: 1619760609,
    owner_id: 572287,
    product_of_interest: {
      name: "chat",
      value: 2,
    },
    properties: {
      company: "sourcefuse",
      product_of_interest: 2,
      use_case: 1,
    },
    protected: 0,
    transaction_pricing_plan: {},
  },
  {
    created: null,
    dates: {
      company: {
        created: 1618297058,
        modified: null,
      },
      product_of_interest: {
        created: 1618297266,
        modified: null,
      },
      use_case: {
        created: 1618297266,
        modified: null,
      },
    },
    email: "xyz@sourcefuse.com",
    id: 571121,
    idp_domain_forbidden: false,
    internal_cipher_key: ",3#(di^t-[?73_#t8:9*qih/s`5/}_0|",
    mau_pricing_plan: {
      pricing_plan: {
        plan_type: "mau_t1",
      },
    },
    modified: 1618297266,
    owner_id: 571170,
    product_of_interest: {
      name: "chat",
      value: 2,
    },
    properties: {
      company: "SourceFuse",
      product_of_interest: 2,
      use_case: 1,
    },
    protected: 0,
    transaction_pricing_plan: {},
  },
  {
    created: null,
    dates: {
      company: {
        created: 1618577901,
        modified: null,
      },
      product_of_interest: {
        created: 1618732591,
        modified: null,
      },
      use_case: {
        created: 1618732591,
        modified: null,
      },
    },
    email: "test@gmail.com",
    id: 571344,
    idp_domain_forbidden: false,
    modified: 1618732591,
    owner_id: 571393,
    product_of_interest: {
      name: "chat",
      value: 2,
    },
    properties: {
      company: "Sourcefuse Technologies",
      product_of_interest: 2,
    },
    protected: 0,
  },
];

export default null;
