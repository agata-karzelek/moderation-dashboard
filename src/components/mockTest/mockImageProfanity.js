export const mockImageModerationOff = {
  imageModerationToggle: "false",
  toolForImageModeration: "sightengine",
  sightengineAPIUser: "236156667",
  sightengineAPIKey: "NSvtoAtDvv8kMDDANGqr",
  sightengineWorkflowId: "wfl_9UyHigiCRqMCDT5wVHLZT",
  sightengineRiskFactorThreshold: "0.25",
  reRouteMessages: "true",
  applyToAllChannelIds: "true",
};

export const mockImageModerationOn = {
  imageModerationToggle: "true",
  toolForImageModeration: "sightengine",
  sightengineAPIUser: "236156667",
  sightengineAPIKey: "NSvtoAtDvv8kMDDANGqr",
  sightengineWorkflowId: "wfl_9UyHigiCRqMCDT5wVHLZT",
  sightengineRiskFactorThreshold: "0.25",
  reRouteMessages: "false",
  applyToAllChannelIds: "true",
};

export const mockImageModerationReoute = {
  imageModerationToggle: "true",
  toolForImageModeration: "sightengine",
  sightengineAPIUser: "236156667",
  sightengineAPIKey: "NSvtoAtDvv8kMDDANGqr",
  sightengineWorkflowId: "wfl_9UyHigiCRqMCDT5wVHLZT",
  sightengineRiskFactorThreshold: "0.25",
  reRouteMessages: "true",
  applyToAllChannelIds: "true",
};

export const mockImageState = {
  applyToAllChannelIds: true,
  initialLoading: false,
  imageModerationToggle: true,
  channelId: "initialText",
  channelIdError: false,
  sightengineAPIUserError: false,
  sightengineAPIKeyError: false,
  sightengineWorkflowIdError: false,
  channelOnChange: false,
  toolForImageModeration: "sightengine",
  sightengineAPIUser: "",
  sightengineAPIKey: "",
  sightengineWorkflowId: "",
  sightengineRiskFactorThreshold: "",
  reRouteMessages: "",
  error: {
    status: false,
    msg: "",
  },
  successMsg: "",
  errorMsg: "",
  successStatus: false,
  errorStatus: false,
  saveLoading: false,
};

export default null;
