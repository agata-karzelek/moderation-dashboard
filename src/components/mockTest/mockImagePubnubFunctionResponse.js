export const mockImagePubnubFunctionResponse = {
  message: "OK",
  payload: [
    {
      cloned_id: null,
      create_user_id: 572002,
      created_date: "Thu, 22 Jul 2021 07:46:04 GMT",
      description: "Image moderation function",
      event_handlers: [
        {
          block_id: 61167,
          channel_groups: null,
          channels: "*",
          code: 'function runProfanity(request){\n             if (request && request.ok) {\n                const bannedChannel = new RegExp("\\\\b(banned)\\\\b", "g")\n                const console = require("console");\n                const query = require("codec/query_string");\n                const xhr = require("xhr");\n                const pubnub = require("pubnub");\n\n                if (bannedChannel.test(request.channels[0])) {\n                  request.message.type = "text";\n                  return request.ok(request);\n                }\n\n                let message = request.message.message;\n                let originalMessage;\n                let moderatedMessage;\n                let imageBannedFlag = false;\n                let imageBannedReouteFlag = false;\n                let textBlockedFlag = false;\n                let textReouteFlag = false;\n                let reasonForModeration;\n\n                const textmoderation = () => {\n                  console.log("start text Moderation Function");\n                  return new Promise((resolve, reject) => {\n    const apiKey = \'d9a83e99c5eb43aaba96ad58be649255\';\n      const abuseTypes = [\'bigotry\', \'personal_attack\', \'criminal_activity\', \'sexual_advances\', \'profanity\'];\n      const serverityTypeObject = {\n        extreme:.25,\n        high:.50,\n        medium:.75,\n        low:1,\n        none:0\n      };\n      const abuseTypeObject = {\n        bigotry:0.5,\n        criminal_activity:0.5,\n        sexual_advances:0.5,\n        personal_attack:0.5,\n        profanity:0.5\n      }\n      const url = "https://api.tisane.ai/parse";\n      const http_options = {\n        "method": "POST",\n        "headers": {\n                    "Ocp-Apim-Subscription-Key": apiKey\n        },\n        "body": JSON.stringify({\n          "language": "en", // or whatever language you use\n          "content": message.text,\n          "settings": {"snippets": true, "format":"dialogue"}\n        }),\n        "timeout" : 5000\n    };\n  \n    return xhr.fetch(url, http_options).then(response => {\n  \n      var body = JSON.parse(response.body);\n      var thirdPartyResponse = (response.status === 200 ? body : { error: body });\n      let checkThresholdForThirdParty = false;\n      const reasons = [];\n      const serverityTypes = [];\n      if (thirdPartyResponse.abuse && Array.isArray(thirdPartyResponse.abuse) && thirdPartyResponse.abuse.length) {\n        thirdPartyResponse.abuse.forEach((element)=>{\n          serverityTypes.push(element.severity);\n          if (abuseTypeObject[element.type] !== 0) {\n            if (abuseTypes.includes(element.type) && abuseTypeObject[element.type] >= serverityTypeObject[element.severity]) {\n              const abuse = element.type.replace(/_/g, \' \');\n              reasons.push(abuse+\'; \'+ element.severity);\n              checkThresholdForThirdParty = true;\n            }\n          }\n        });\n      }\n    if(checkThresholdForThirdParty){\n        message.text = message.text.replace(/[a-z-A-Z-!]/g, \'*\');\n        return resolve(true);\n    }\n    message.type = "text"\n    return resolve(true);\n    }).catch(err => {\n      var thirdPartyResponse = { error: err };\n      Object.assign(message, { thirdPartyResponse });\n      return reject(message);\n  })\n   })\n                };\n\n                if (request.message.file) {\n                      var apiUrl = "https://api.sightengine.com/1.0/check-workflow.json";\n                      var fileUrl = "";\n                      const channel = request.channels[0];\n                      const fileId = request.message.file.id;\n                      const fileName = request.message.file.name;\n                      fileUrl = pubnub.getFileUrl({\n                        channel: channel,\n                        id: fileId,\n                        name: fileName,\n                      });\n                      const queryParams = {\n                        \'api_secret\': \'NSvtoAtDvv8kMDDANGqr\',\n                        \'workflow\': \'wfl_9UyHigiCRqMCDT5wVHLZT\',\n                        \'api_user\': \'236156667\',\n                        url: fileUrl\n                      };\n\n                      const imageModerationFunction = () => {\n                        return new Promise((resolve, reject) => {\n                          console.log("start image Moderation Function");\n                          xhr\n                            .fetch(apiUrl + "?" + query.stringify(queryParams))\n                            .then(function (r) {\n                              const body = JSON.parse(r.body || r);\n                              return resolve(body);\n                            })\n                            .catch(function (e) {\n                              console.error(e);\n                              return reject(e);\n                            });\n                        });\n                      };\n\n                      return Promise.all([imageModerationFunction(), textmoderation()])\n                        .then((values) => {\n                          console.log("values", values);\n                          let payload = {};\n                          if (\n                            values[0] &&\n                            values[0].summary &&\n                            values[0].summary.reject_prob &&\n                            \'0.25\' < values[0].summary.reject_prob\n                          ) {\n                               imageBannedReouteFlag = true;\n                            }\n\n                          if (imageBannedReouteFlag) {\n                            payload = {\n                              type: "image",\n                              image: fileUrl,\n                              reason: values[0].summary.reject_reason,\n                            }\n                          }\n\n                          if (textReouteFlag) {\n                            payload.type = "text";\n                            if (originalMessage) {\n                              payload.originalMessage = originalMessage;\n                            } else {\n                              payload.originalMessage = request.message.message.text;\n                            }\n                            if (moderatedMessage) {\n                              payload.moderatedMessage = moderatedMessage;\n                            }\n                            if (reasonForModeration) {\n                              payload.reason = reasonForModeration;\n                            }\n                          }\n\n                          if (payload.type) {\n                            return pubnub\n                            .publish({\n                              channel: "banned." + request.channels[0],\n                              message: payload,\n                            })\n                            .then((publishResponse) => {\n                              if (textBlockedFlag && imageBannedFlag) {\n                                return request.abort("moderated message");\n                              }\n                              if (!imageBannedFlag && !imageBannedReouteFlag) {\n                                message.file = {};\n                                message.file.url = fileUrl;\n                              }\n                              if (textBlockedFlag) {\n                                delete message.text;\n                              }\n                              return request.ok(message);\n                            })\n                            .catch((err) => {\n                              console.error(err);\n                            });\n                          }\n                          if (values[0] || values[1]) {\n                           if (textBlockedFlag && imageBannedFlag) {\n                            return request.abort("moderated message");\n                           }\n                            if (textBlockedFlag) {\n                              delete message.text;\n                            }\n                            if (!imageBannedFlag) {\n                              request.message.message.file = {};\n                              request.message.message.file.url = fileUrl;\n                              request.message.message.type = "text";\n                            }\n                            return request.ok(message);\n                          }\n                        })\n                        .catch((err) => {\n                          console.log(err);\n                          return request.abort(err);\n                        });\n                    }\n              }\n                \n        return {\n          imageModerationToggle: \'true\',\n          toolForImageModeration: \'sightengine\',\n          sightengineAPIUser: \'236156667\',\n          sightengineAPIKey: \'NSvtoAtDvv8kMDDANGqr\',\n          sightengineWorkflowId: \'wfl_9UyHigiCRqMCDT5wVHLZT\',\n          sightengineRiskFactorThreshold: \'0.25\',\n          reRouteMessages: \'true\',\n          applyToAllChannelIds: \'true\'\n        }\n    }',
          create_user_id: 572002,
          created_date: "Thu, 22 Jul 2021 07:46:10 GMT",
          event: "js-before-publish-file",
          id: 64668,
          log_level: "debug",
          modified_date: "Wed, 28 Jul 2021 10:37:41 GMT",
          modified_user_id: 572002,
          name: "BLOCK-61157-IMAGE-MODERATION",
          order_index: 0,
          output: "output-0.5823105582419438",
          path: "",
          rate: 0,
          state: "running",
          status: 1,
          test_payload:
            '{"text":"case:3:1: Normal image reouting and text normal banned and reouting shitt!"}',
          type: "js",
          wildcard_publish_channel: "automationImage",
        },
      ],
      icon_url: null,
      id: 61165,
      intended_state: "running",
      key_id: 1142278,
      modified_date: "Wed, 28 Jul 2021 10:37:45 GMT",
      modified_user_id: 572402,
      name: "KEY-1042218-IMAGE-MODERATION",
      pub_key: "pub-c-d29deb19-58c1-49c2-887d-ef3131d451f8",
      state: "running",
      state_change_date: "Wed, 29 Jul 2021 10:37:45 GMT",
      status: 1,
      sub_key: "sub-c-449acf00-eaba-12eb-b05e-3ebc6f27b518",
      template: 0,
    },
    {
      cloned_id: null,
      create_user_id: 572102,
      created_date: "Thu, 20 Jul 2021 13:42:21 GMT",
      description: "This is a profanity function",
      event_handlers: [
        {
          block_id: 61072,
          channel_groups: null,
          channels: "*",
          code: "function runProfanity(request){\n     if(request && request.ok){\n    const xhr = require(\"xhr\");\n    const console = require('console');\n    var bannedChannel = new RegExp(\"\\\\b(banned)\\\\b\", \"g\")\n    let message = request.message;\n  \n    if(bannedChannel.test(request.channels[0])){\n       request.message.type = \"text\";\n       return request.ok(message);\n     }\n  \n     const apiKey = 'd9a83e99c5eb43aaba96ad58be649255';\n      const abuseTypes = ['bigotry', 'personal_attack', 'criminal_activity', 'sexual_advances', 'profanity'];\n      const serverityTypeObject = {\n        extreme:.25,\n        high:.50,\n        medium:.75,\n        low:1,\n        none:0\n      };\n      const abuseTypeObject = {\n        bigotry:0.5,\n        criminal_activity:0.5,\n        sexual_advances:0.5,\n        personal_attack:0.5,\n        profanity:0.5\n      }\n      const url = \"https://api.tisane.ai/parse\";\n      const http_options = {\n        \"method\": \"POST\",\n        \"headers\": {\n                    \"Ocp-Apim-Subscription-Key\": apiKey\n        },\n        \"body\": JSON.stringify({\n          \"language\": \"en\", // or whatever language you use\n          \"content\": message.text,\n          \"settings\": {\"snippets\": true, \"format\":\"dialogue\"}\n        }),\n        \"timeout\" : 5000\n    };\n  \n    return xhr.fetch(url, http_options).then(response => {\n  \n      var body = JSON.parse(response.body);\n      var thirdPartyResponse = (response.status === 200 ? body : { error: body });\n      let checkThresholdForThirdParty = false;\n      const reasons = [];\n      const serverityTypes = [];\n      if (thirdPartyResponse.abuse && Array.isArray(thirdPartyResponse.abuse) && thirdPartyResponse.abuse.length) {\n        thirdPartyResponse.abuse.forEach((element)=>{\n          serverityTypes.push(element.severity);\n          if (abuseTypeObject[element.type] !== 0) {\n            if (abuseTypes.includes(element.type) && abuseTypeObject[element.type] >= serverityTypeObject[element.severity]) {\n              const abuse = element.type.replace(/_/g, ' ');\n              reasons.push(abuse+'; '+ element.severity);\n              checkThresholdForThirdParty = true;\n            }\n          }\n        });\n      }\n    if(checkThresholdForThirdParty){\n               message.text = message.text.replace(/[a-z-A-Z-!]/g, '*');\n              return request.ok(message);\n          }\n         request.message.type = \"text\"\n         return request.ok(message);\n  \n     }).catch(err => {\n         var thirdPartyResponse = { error: err };\n         Object.assign(message, { thirdPartyResponse });\n  \n         return request.ok(message);\n     });\n      }\n     return {\n       wordListProfanity: 'false',\n       automaticProfanity: 'true',\n       textModerationToggle: 'true',\n       wordList:{\n         applyToAllChannelIdsWordlist:'true',\n         wordListReRouteMessages: 'true',\n         wordListModType: 'Block-message',\n         wordListCharacterToMaskWith:'*',\n         englishProfanity:'anal|anus|arse|ass|ballsack|balls|bastard|bitch|biatch|bloody|blowjob|blow job|bollock|bollok|boner|boob|bugger|bum|butt|buttplug|clitoris|cock|coon|crap|cunt|damn|dick|dildo|dyke|fag|feck|fellate|fellatio|felching|fuck|f u c k|fudgepacker|fudge packer|flange|Goddamn|God damn|hell|homo|jerk|jizz|knobend|knob end|labia|lmao|lmfao|muff|nigger|nigga|penis|piss|poop|prick|pube|pussy|queer|scrotum|sex|shit|s hit|sh1t|slut|smegma|spunk|tit|tosser|turd|twat|vagina|wank|whore|wtf',\n         hindiProfanity: '',\n         frenchProfanity: '',\n         spanishProfanity: '',\n         portugeseProfanity: ''\n       },\n       automaticDetection:{\n         applyToAllChannelIdsAutomatic: 'true',\n         automaticDetectionReRouteMessages: 'false',\n         automaticDetectionModType: 'mask-message',\n         automaticDetectionCharacterToMaskWith:'*',\n         toolForAutomaticDetection:'tisane',\n         siftNinjaRiskFactorThresholdVulgar:'0',\n         siftNinjaRiskFactorThresholdSexting:'0',\n         siftNinjaRiskFactorThresholdRacism:'0',\n         siftNinjaAccountName:'1',\n         siftNinjaChannelName:'sa',\n         siftNinjaApiKey:'aaa',\n         tisaneRiskFactorThresholdBigotry:'0.5',\n         tisaneRiskFactorThresholdCyberBullying:'0.5',\n         tisaneRiskFactorThresholdCriminalActivity:'0.5',\n         tisaneRiskFactorThresholdSexualAdvances:'0.5',\n         tisaneRiskFactorThresholdProfanity:'0.5',\n         tisaneApiKey:'d9a83e99c5eb43aaba96ad58be649255',\n         tisaneLanguage:'English'\n       },\n     }\n  }",
          create_user_id: 571002,
          created_date: "Thu, 27 Jul 2021 13:42:26 GMT",
          event: "js-before-publish",
          id: 64613,
          log_level: "debug",
          modified_date: "Wed, 28 Jul 2021 10:37:27 GMT",
          modified_user_id: 572002,
          name: "BLOCK-61172",
          order_index: 0,
          output: "output-0.5823105682419438",
          path: "",
          rate: 0,
          state: "running",
          status: 1,
          test_payload: '{"text":"Hello World! anus"}',
          type: "js",
          wildcard_publish_channel: "hi",
        },
      ],
      icon_url: null,
      id: 61172,
      intended_state: "running",
      key_id: 1042278,
      modified_date: "Wed, 28 Jul 2021 10:37:30 GMT",
      modified_user_id: 572002,
      name: "KEY-1042278",
      pub_key: "pub-c-d29deb49-58c1-49c2-887d-ef3131d451f8",
      state: "running",
      state_change_date: "Wed, 28 Jul 2021 10:37:30 GMT",
      status: 1,
      sub_key: "sub-c-449acf00-eaba-11eb-b05e-3ebc6f27b518",
      template: 0,
    },
  ],
  status: 200,
};

export default null;
