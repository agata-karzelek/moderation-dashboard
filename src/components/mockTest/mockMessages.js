export const mockMessages = [
  {
    channel: "pro.one",
    timetoken: "16287753052976351",
    message: {
      message: {
        text: "hallelujah",
      },
      type: "text",
      file: {},
      time: "07:05",
      timetoken: "16287753052976351",
      text: "hallelujah",
    },
    messageType: 4,
    uuid: "pn-6f3228b8-32d3-4921-b9eb-af8e96bbde8d",
  },
  {
    channel: "pro.one",
    timetoken: "16287765563258261",
    message: {
      message: {
        text: "hallelujah",
      },
      type: "text",
      file: {},
      time: "07:25",
      timetoken: "16287765563258261",
      text: "hallelujah",
    },
    messageType: 4,
    uuid: "pn-6f3228b8-32d3-4921-b9eb-af8e96bbde8d",
  },
  {
    channel: "pro.one",
    timetoken: "16287765658563945",
    message: {
      message: {
        text: "hallelujah",
      },
      type: "text",
      file: {
        url: "https://ps.pndsn.com/v1/files/sub-c-1335d6e8-a993-11eb-9cd5-b6e2b128ec2a/channels/pro.one/files/15bef783-2c5d-43f1-942a-52894d5246fb/self1.jpeg",
        id: "15bef783-2c5d-43f1-942a-52894d5246fb",
        name: "self1.jpeg",
      },
      time: "07:26",
      timetoken: "16287765658563945",
      text: "hallelujah",
      image:
        "https://ps.pndsn.com/v1/files/sub-c-1335d6e8-a993-11eb-9cd5-b6e2b128ec2a/channels/pro.one/files/15bef783-2c5d-43f1-942a-52894d5246fb/self1.jpeg",
    },
    messageType: 4,
    uuid: "pn-6f3228b8-32d3-4921-b9eb-af8e96bbde8d",
  },
  {
    channel: "pro.one",
    timetoken: "16287766010372679",
    message: {
      message: {
        text: "hallelujah",
      },
      type: "text",
      file: {
        url: "https://ps.pndsn.com/v1/files/sub-c-1335d6e8-a993-11eb-9cd5-b6e2b128ec2a/channels/pro.one/files/6205f5c3-3da9-41ba-bb9c-e5e49e61aedd/self1.jpeg",
        id: "6205f5c3-3da9-41ba-bb9c-e5e49e61aedd",
        name: "self1.jpeg",
      },
      time: "07:26",
      timetoken: "16287766010372679",
      text: "hallelujah",
      image:
        "https://ps.pndsn.com/v1/files/sub-c-1335d6e8-a993-11eb-9cd5-b6e2b128ec2a/channels/pro.one/files/6205f5c3-3da9-41ba-bb9c-e5e49e61aedd/self1.jpeg",
    },
    messageType: 4,
    uuid: "pn-6f3228b8-32d3-4921-b9eb-af8e96bbde8d",
  },
];

export default null;
