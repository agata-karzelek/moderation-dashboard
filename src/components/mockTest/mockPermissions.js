export const mockPermissions = {
  "account-571121": {
    account_id: 571121,
    entity_id: 571121,
    entity_type: "account",
    internal: 0,
    name: "Account Admin",
    permission: "write",
    scopes: {
      account_id: "write",
      app_id: "write",
      block_id: "write",
      event_handler_id: "write",
      key_id: "write",
    },
  },
  "app-35384471": {
    account_id: 571344,
    entity_id: 35384471,
    entity_type: "app",
    internal: 0,
    name: "App Admin",
    permission: "write",
    scopes: {
      app_id: "write",
      key_id: "write",
    },
  },
  "app-35386912": {
    account_id: 572023,
    entity_id: 35386912,
    entity_type: "app",
    internal: 0,
    name: "App Admin",
    permission: "write",
    scopes: {
      app_id: "write",
      key_id: "write",
    },
  },
};

export default null;
