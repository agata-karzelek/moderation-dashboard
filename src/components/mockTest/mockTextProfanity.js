const createMockObject = ({
  wordListProfanity = "true",
  automaticProfanity = "true",
  textModerationToggle = "true",
  wordListReRouteMessages = "true",
  wordListModType = "Block-message",
  automaticDetectionReRouteMessages = "true",
  automaticDetectionModType = "block-message",
}) => {
  return {
    wordListProfanity,
    automaticProfanity,
    textModerationToggle,
    profanityList: {
      English:
        "anal|anus|arse|ass|ballsack|balls|bastard|bitch|biatch|bloody|blowjob|blow job|bollock|bollok|boner|boob|bugger|bum|butt|buttplug|clitoris|cock|coon|crap|cunt|damn|dick|dildo|dyke|fag|feck|fellate|fellatio|felching|fuck|f u c k|fudgepacker|fudge packer|flange|Goddamn|God damn|hell|homo|jerk|jizz|knobend|knob end|labia|lmao|lmfao|muff|nigger|nigga|penis|piss|poop|prick|pube|pussy|queer|scrotum|sex|shit|s hit|sh1t|slut|smegma|spunk|tit|tosser|turd|twat|vagina|wank|whore|wtf",
      Hindi: "",
      Portugese: "",
      French: "",
      Spanish: "",
    },
    wordList: {
      applyToAllChannelIdsWordlist: "true",
      wordListReRouteMessages,
      wordListModType,
      wordListCharacterToMaskWith: "*",
      englishProfanity:
        "anal|anus|arse|ass|ballsack|balls|bastard|bitch|biatch|bloody|blowjob|blow job|bollock|bollok|boner|boob|bugger|bum|butt|buttplug|clitoris|cock|coon|crap|cunt|damn|dick|dildo|dyke|fag|feck|fellate|fellatio|felching|fuck|f u c k|fudgepacker|fudge packer|flange|Goddamn|God damn|hell|homo|jerk|jizz|knobend|knob end|labia|lmao|lmfao|muff|nigger|nigga|penis|piss|poop|prick|pube|pussy|queer|scrotum|sex|shit|s hit|sh1t|slut|smegma|spunk|tit|tosser|turd|twat|vagina|wank|whore|wtf",
      hindiProfanity: "",
      frenchProfanity: "",
      spanishProfanity: "",
      portugeseProfanity: "",
    },
    automaticDetection: {
      applyToAllChannelIdsAutomatic: "true",
      automaticDetectionReRouteMessages,
      automaticDetectionModType,
      automaticDetectionCharacterToMaskWith: "*",
      toolForAutomaticDetection: "tisane",
      siftNinjaRiskFactorThresholdVulgar: "0",
      siftNinjaRiskFactorThresholdSexting: "0",
      siftNinjaRiskFactorThresholdRacism: "0",
      siftNinjaAccountName: "1",
      siftNinjaChannelName: "sa",
      siftNinjaApiKey: "aaa",
      tisaneRiskFactorThresholdBigotry: "0.5",
      tisaneRiskFactorThresholdCyberBullying: "0.5",
      tisaneRiskFactorThresholdCriminalActivity: "0.5",
      tisaneRiskFactorThresholdSexualAdvances: "0.5",
      tisaneRiskFactorThresholdProfanity: "0.5",
      tisaneApiKey: "d9a83e99c5eb43aaba96ad58be649255",
      tisaneLanguage: "English",
    },
  };
};

export const mockTextProfanityOff = createMockObject({
  automaticProfanity: "false",
  textModerationToggle: "false",
});

export const mockMasktWordProfanityOn = createMockObject({
  automaticProfanity: "false",
  wordListReRouteMessages: "false",
  wordListModType: "Mask-word",
});

export const mockMaskReouteWordProfanityOn = createMockObject({
  automaticProfanity: "false",
  wordListModType: "Mask-word",
});

export const mockBannedWordProfanityOn = createMockObject({
  automaticProfanity: "false",
  wordListReRouteMessages: "false",
});

export const mockBannedReouteWordProfanityOn = createMockObject({
  automaticProfanity: "false",
});

export const mockBlockAutomationOn = createMockObject({
  wordListProfanity: "false",
  automaticDetectionReRouteMessages: "false",
});

export const mockBlockReoutingAutomationOn = createMockObject({
  wordListProfanity: "false",
});

export const mockMaskReoutingAutomationOn = createMockObject({
  wordListProfanity: "false",
  automaticDetectionModType: "mask-message",
});

export const mockMaskAutomationOn = createMockObject({
  wordListProfanity: "false",
  automaticDetectionModType: "mask-message",
  automaticDetectionReRouteMessages: "false",
});

export const mockWithoutMaskandBannedAutoOn = createMockObject({
  textModerationToggle: "false",
  automaticProfanity: "false",
  wordListModType: "Block-message-1",
  automaticDetectionModType: "mask-message",
  automaticDetectionReRouteMessages: "false",
});

export const mockTextModerationProps = (defaultTool = "tisane") => ({
  automaticDetection: {
    applyToAllChannelIdsAutomatic: true,
    toolForAutomaticDetection: defaultTool,
    siftNinjaRiskFactorThresholdVulgar: 0.5,
    siftNinjaRiskFactorThresholdSexting: 0.5,
    siftNinjaRiskFactorThresholdRacism: 0.5,
    siftNinjaAccountName: "",
    siftNinjaChannelName: "",
    siftNinjaApiKey: "",
    tisaneRiskFactorThresholdBigotry: 0.5,
    tisaneRiskFactorThresholdCyberBullying: 0.5,
    tisaneRiskFactorThresholdCriminalActivity: 0.5,
    tisaneRiskFactorThresholdSexualAdvances: 0.5,
    tisaneRiskFactorThresholdProfanity: 0.5,
    tisaneApiKey: "",
    tisaneLanguage: "Hindi",
    automaticDetectionChannel: "*",
    automaticChannelError: false,
    automaticMaskCharError: false,
    automaticDetectionReRouteMessages: false,
    automaticDetectionModType: "mask-message",
    automaticDetectionCharacterToMaskWith: "*",
  },
  textModerationToggle: false,
  wordListProfanity: false,
  automaticProfanity: false,
  channelOnChange: false,
  saveLoading: false,
  initialLoading: true,
  errorStatus: false,
  successStatus: false,
  errorMsg: "",
  successMsg: "",
});

export default null;
