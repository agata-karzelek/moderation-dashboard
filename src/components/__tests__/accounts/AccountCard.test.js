import React from "react";
import { shallow } from "enzyme";
import AccountCard from "../../accounts/AccountCard";
import { mockAccounts } from "../../mockTest/mockAccounts";

const clickFn = jest.fn();

describe("Account Card Test Cases", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <AccountCard account={mockAccounts[1]} index={1} onClick={clickFn} handleClick={clickFn} />
    );
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Check Value", () => {
    expect(wrapper.find('[testid="capitalChar"]').getElements()[0].props.children).toBe("Xy");
    expect(wrapper.find('[testid="email"]').getElements()[0].props.children).toBe(
      "xyz@sourcefuse.com"
    );
  });

  test("on click", () => {
    wrapper.find('[testid="1"]').getElements()[0].props.onClick(clickFn);
    expect(clickFn).toHaveBeenCalledTimes(1);
    expect(wrapper.find('[testid="capitalChar"]').getElements()[0].props.children).toBe("Xy");
  });
});
