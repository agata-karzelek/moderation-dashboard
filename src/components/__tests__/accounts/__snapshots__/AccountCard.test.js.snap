// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Card Test Cases Snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(MenuItem))
    onClick={[Function]}
    testid="1"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={2}
        xs={2}
      >
        <WithStyles(ForwardRef(Avatar))
          className="makeStyles-avatar-5"
          style={
            Object {
              "backgroundColor": "#CAF0FF",
              "borderRadius": "8px",
              "color": "#3BC8FF",
              "height": "30px",
              "width": "30px",
            }
          }
          variant="square"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-avatarText-4"
            testid="capitalChar"
          >
            Xy
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={9}
        xs={9}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-email-3"
          noWrap={true}
          testid="email"
        >
          xyz@sourcefuse.com
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={1}
        xs={1}
      >
        <Memo(ForwardRef(CheckCircleIcon))
          className="makeStyles-checkIcon-2"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(MenuItem))>
</Fragment>
`;
